Simmicmenten

taustinn11 opened the pull request:
Completed the simmicmenten equation. It includes the hill slope feature (default is h = 1). This will allow it to be used for Michaelis-Menten modeling with or without cooperativity.

Future version should include some feature to extract the regressed Vmax value such that it can be used to derive kcat

Commit:
Added sandbox.rmd to Rbuildignore to keep out of the package

started with sandbox
working on one-site total binding Eq
Having to troubleshoot log transformation

Commit:
Can coerce graphing of log = T values by using loess method with y ~ x formula.

However, going to need to try different values as start arguments and see if we can get nlsLM to work

Commit:
Added comments.

Importantly, the loess method starts to look very poor as you increase the cv (obviously). This cannot be used.

Commit:
Use stat_function call to bypass nlsLM error. Tested with higher values and higher variance and seems to still plot well

Generated a .R file for it using use_r()

Commit:
Updated sim1sbind to plot a stat_function based on parameters gathered from nls call in the log-linear -> coefficients hack

Vast improvement because now stat_function call plots a line that responds to the variation in simulated points

Commit:
quick fix

Commit:
Added sandbox.rmd to .gitignore - keeping for local to test

Commit:
Starting simmicmenten

Used check() for sim1sbind to find errors. It all worked, so it updated the man. Ready for push

Commit:
Committing before clearing cached to remove sandbox.rmd

Commit:
.gitignore fix

Commit:
.gitignore fix

Commit:
Generated simmicmenten.R w/ use_r()

Commit:
Filled out description of simmicmenten.R

Commit:
Quick check after writing roxygen template for simmicmenten

Found error. Correcting after this commit

Commit:
corrected log = F to log = FALSE as directed to by check()

Commit:
Check() returned 0 errors, 0 warnings, and 0 notes.

Updated man/simmicmenten.Rd as a result

TJMurphy commented on R/simmicmenten.R:
```diff
@@ -0,0 +1,42 @@
+simmicmenten <- function(x, vmax, km, cv, reps, h = 1, log=F) {
```
How about a different name?: simmm, simmicmen?

taustinn11 replied:
simmicmen seems fine to me. simmm seems a little too vague.

taustinn11 replied:
Honestly, I'm not quite sure how to do this. I could create a new roxygen document and all of that by generating a new use_r() call, but then I'm not quite sure how to properly get rid of the old function.

I'm going to leave it as is for now until I have some more time later to change the name to simmicmen

TJMurphy commented on R/simmicmenten.R:
```diff
@@ -0,0 +1,42 @@
+      log10(x)} else {
+        x}, y)) +
+    ggplot2::geom_point(size=2) +
+    ggplot2::labs(title="model: y=vmax*x^h/(x+km^h)") +
```
add x-axis label: x="x"

taustinn11 replied:
Done!

TJMurphy commented on R/simmicmenten.R:
```diff
@@ -0,0 +1,42 @@
+    ggplot2::geom_point(size=2) +
+    ggplot2::labs(title="model: y=vmax*x^h/(x+km^h)") +
+    if (log) {
+      ggplot2::stat_function(geom = "smooth", fun = function(x) y = vmax.var*10^x^h/(km.var^h + 10^x),
```
We run into the same issue as with sim1sbind. Maybe it is better to have a function dedicated to log, and a separate function dedicated to linear? 

There is this old saying from Jared, "write functions that do one thing well. try not to make them do two things"

taustinn11 replied:
I really do think that this will work for both. I left a more detailed comment about this over on the sim1sbind pull request thread. 

I understand what you mean by having a function do two things, but I think that choosing between a log-transformed x vector or a linear x vector doesn't seem like this function (or sim1sbind) is trying to do too much

TJMurphy commented on R/simmicmenten.R:
```diff
@@ -0,0 +1,42 @@
+
+  for (i in 1) {
+    if (log) {
+      model <- stats::lm(log(y) ~ log(x), data=values)
```
No. there is no need to log y values. Only the x value gets logged.

taustinn11 replied:
These log values are just for the nls() call right after. Log y does not get plotted. I described this in more detail on the sim1sbind pull request thread.

Commit:
Altered some wording of params (as per TJ request)

Commit:
Changed title

Clarified labels in if (log) conditionals

Commit:
Reformatted to have multiple if conditionals - should fix error about adding ggprotos together

Commit:
Stored ggplot calls (with conditionals) into objects and then printed them at the end

Allows for better labeling of the axes as far as I can figure out right now

Commit:
Found a workaround for programming the labels and geom_smooth calls for if (log). It now returns the ggplot call without needing to add objects together or use a print command

Commit:
Check() after ensuring function works appropriately

Commit:
Merge branch 'sim1sbind' of https://github.com/TJMurphy/nlfitr into simmicmenten

Commit:
Fixed labels for log = T or F; used strategy from sim1sbind(logplot and linplot)

Commit:
check() after fixing simmicmenten

This is now updating the man/sim1sbind.Rd file on this branch to include the changes I made on sim1sbind branch.

Commit:
modified @param x to be more accurate

added example using exponential values for x

changed stats::nls to minpack.lm::nlsLM for log = T parameter estimation

Commit:
quick check()

modified logdose in @examples to not be too long (gave be a note with check())

Commit:
Merge branch 'sim1sbind' of https://github.com/TJMurphy/nlfitr into simmicmenten

Commit:
Switched stats::nls for minpack.lm::nlsLM

Played around with using log = TRUE and h !=1. **Encountered issues that will need to be worked around**

Amended the description to include a workaround for this. But this will need to be revisited.

TJMurphy committed:
converts from log base e to log base 10
